utils/fixed_decimal/src/decimal.rs

```diff
@@ -178,6 +179,64 @@ impl FixedDecimal {
         Ok(result)
     }
 
+    /// Initialize a `FixedDecimal` with a float value, using Ryu under the hood
+    /// to get digits from the value. The `post_decimal_places` parameter places
+    /// a limit on the number of digits after the decimal point.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use fixed_decimal::FixedDecimal;
+    ///
+    /// let dec = FixedDecimal::from_float_ryu(12.345, 2).unwrap();
+    /// assert_eq!("12.34", dec.to_string());
+    /// ```
+    #[cfg(feature = "ryu_decimal")]
+    pub fn from_float_ryu(value: f32, post_decimal_places: u16) -> Result<Self, Error> {
+        let ryu_decimal = ryu_floating_decimal::f2d(value);
+        let mut limited_mantissa = ryu_decimal.mantissa;
+        let mut limited_exponent: i16 = ryu_decimal.exponent.try_into().map_err(|_| Error::Limit)?;
+        if limited_exponent < 0 {
+            while -limited_exponent as u16 > post_decimal_places {
+                limited_exponent += 1;
+                limited_mantissa /= 10;
+            }
+        }
+
+        let mut res = FixedDecimal::from(limited_mantissa);
+        res.multiply_pow10(limited_exponent)?;
+        Ok(res)
+    }
+
+    /// Initialize a `FixedDecimal` with a double value, using Ryu under the hood
+    /// to get digits from the value. The `post_decimal_places` parameter places
+    /// a limit on the number of digits after the decimal point.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use fixed_decimal::FixedDecimal;
+    ///
+    /// let dec = FixedDecimal::from_double_ryu(12.345, 2).unwrap();
+    /// assert_eq!("12.34", dec.to_string());
+    /// ```
+    #[cfg(feature = "ryu_decimal")]
+    pub fn from_double_ryu(value: f64, post_decimal_places: u16) -> Result<Self, Error> {
+        let ryu_decimal = ryu_floating_decimal::d2d(value);
+        let mut limited_mantissa = ryu_decimal.mantissa;
+        let mut limited_exponent: i16 = ryu_decimal.exponent.try_into().map_err(|_| Error::Limit)?;
+        if limited_exponent < 0 {
+            while -limited_exponent as u16 > post_decimal_places {
+                limited_exponent += 1;
+                limited_mantissa /= 10;
+            }
+        }
+
+        let mut res = FixedDecimal::from(limited_mantissa);
+        res.multiply_pow10(limited_exponent)?;
+        Ok(res)
+    }
+
```

Comment:
Could you add tests to ensure that trailing zeros work properly (like "1.20")?

Reply:
Just to make sure I understand correctly: we should use `post_decimal_places` as the _exact_ number of places after the point instead of a limit? So that `from_double_ryu(1.2, 2)` would result in "1.20".

Reply:
Originally I was thinking only a *minimum* number of decimal places, but I'm also okay with it being the exact number (both minimum and maximum).

Reply:
Exact ended up being a bit more intuitive for situations like representing `1234`, which would otherwise have a `.0` appended automatically regardless of the minimum. So went ahead and implemented that along with some additional tests.